sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -100,3 +102,19 @@
 // clang emits "warning:" in the message pragma output
 #define __SYCL_WARNING(msg) __pragma(message(msg))
 #endif // __GNUC__
+
+// defined __SYCL_UNROLL to add pragma/attribute unroll to a loop
+#ifndef __SYCL_UNROLL
+#if defined(__clang__)
+#define __SYCL_UNROLL(x) _Pragma(STRINGIFY(unroll_count x))
+#elif defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER)
+#define __SYCL_UNROLL(x) _Pragma(STRINGIFY(unroll x))
+#elif (defined(__GNUC__) && __GNUC__ >= 8) ||                                  \
+    (defined(__GNUG__) && __GNUG__ >= 8)
+#define __SYCL_UNROLL(x) _Pragma(STRINGIFY(GCC unroll x))
+#else
+#define __SYCL_UNROLL(x)
+#endif // compiler switch
+#endif // __SYCL_UNROLL
+
+#endif // STRINGIFY
```

Comment:
Any reason to scope so many other macros on whether `STRINGIFY` was previously defined or not?

Also, I would name it `__STRINGIFY` so it doesn't clash with user-defined macros and I would `#undef` it when you're done using it because we don't want people to rely on accessor.hpp giving them this macro as part of its interface.

Reply:
Renamed `STRINGIFY` and moved it a bit later. TBH I think that `__STRINGIFY` can be used again at some point of time in defines_elementary.hpp .

Regarding `#undef` - well, it's a valid comment, but surprisingly it doesn't work, see: https://godbolt.org/z/75nrETGzf

Reply:
So yes,

> but surprisingly it doesn't work

Basically defines_elementary.hpp  is a helper file used across SYCL headers, including accessor.hpp